Modules/IO/TIFF/src/itkTIFFImageIO.cxx

```diff
@@ -559,24 +555,24 @@ TIFFImageIO::Write(const void * buffer)
 void
 TIFFImageIO::InternalWrite(const void * buffer)
 {
-  const auto * outPtr = (const char *)buffer;
+  const auto * outPtr = static_cast<const char *>(buffer);
 
-  unsigned int page, pages = 1;
+  uint16 page, pages = 1;
 
   const SizeValueType width = m_Dimensions[0];
   const SizeValueType height = m_Dimensions[1];
   if (m_NumberOfDimensions == 3)
   {
-    pages = m_Dimensions[2];
+    pages = static_cast<uint16>(m_Dimensions[2]);
   }
 
-  int  scomponents = this->GetNumberOfComponents();
-  auto resolution_x = static_cast<float>(m_Spacing[0] != 0.0 ? 25.4 / m_Spacing[0] : 0.0);
-  auto resolution_y = static_cast<float>(m_Spacing[1] != 0.0 ? 25.4 / m_Spacing[1] : 0.0);
+  uint16 scomponents = static_cast<uint16>(this->GetNumberOfComponents());
```

Comment:
(Minor) You could use `auto`, as the type is clearly redundant here. (Suggested by Visual Assist Code Inspection).